settings/movement.py

```diff
@@ -0,0 +1,169 @@
+from data.movement import MovementActions
+from memory.space import Allocate, Bank, Reserve, Read
+import instruction.asm as asm
+
+WALK_SPEED = 2
+SPRINT_SPEED = 3
+DASH_SPEED = 4
+
+class Movement:
+    def __init__(self):
+        import args
+        self.movement = args.movement or MovementActions.AUTO_SPRINT
+
+        if self.movement != MovementActions.ORIGINAL:
+            self.mod()
+
+    def mod(self):
+        length = 0
+        src = []
+
+        (length, src) = self.get_auto_sprint_src()
+
+        space = Allocate(Bank.F0, length, "Sprint subroutine")
+        space.write(src)
+
+        src = [
+            asm.JSL(space.start_address_snes),
+        ]
+        space = Reserve(0x04e21, 0x04e37, "auto sprint", asm.NOP())
+        space.write(src)
+
+        self.sliding_dash_fix()
+
+
+    def get_auto_sprint_src(self):
+        import args
+        CURRENT_MAP_BYTE = 0x82 # 2 bytes
+        OWZERS_MANSION_ID = 0x0D1 # the door room can create visual artifacts on the map while dashing
+        CONTROLLER1_BYTE2 = 0x4219
+        SPRINT_SHOES_BYTE = 0x11df
+        SPRINT_SHOES_MASK = 0x20
+        B_BUTTON_MASK = 0x80
+        FIELD_RAM_SPEED = 0x0875
+
+        # moving at dash speed in Owzer's door room, or carrying it out via the door glitch will cause graphical artifacting randomly.
+        # Simply disabling B button in Owzers to keep it consistent in WC. Will not worry about the door glitch
+        src = [
+            # Need a sanity check here - This is causing an issue. Go to most maps and you will be unable to move south
```

Comment:
I'll take a look at this; I think we'll want this fixed before merging to main.

Reply:
Which maps? I enabled this (with changing the Allocate call to the Write call, since I didnt' want to recalculate the length) and didn't experience this issue.

Reply:
Sorry, specifically I have noticed it in Albrook and Imperial base (before sealed gate)